jwst/extract_2d/nirspec.py

```diff
@@ -82,6 +84,10 @@ def nrs_extract2d(input_model, slit_name=None):
             # The overall subarray offset is recorded in model.meta.subarray.
             set_slit_attributes(new_model, slit, xlo, xhi, ylo, yhi)
 
+            if (new_model.meta.exposure.type.lower() == 'nrs_fixedslit') and \
+                (slit.name == input_model.meta.instrument.fixed_slit):
+                get_source_xpos(new_model)
+
```

Comment:
> The other checks in the wavecorr step can then just be removed - we won't need to check if it is the primary slit or defined via the MSA file, since that's now handled here. We just need to check if it's a point source.

Agree RE the checks for dither info - those should be in `extract_2d` - but it appears that for MOS/FS combined processing [this `for` loop](https://github.com/spacetelescope/jwst/blob/95c8eb5ce89288f3ca056417c31c0bc527329f44/jwst/wavecorr/wavecorr.py#L77) is still needed to iterate over all slits, and therefore the check for whether it's the primary slit is still needed.  Am I understanding that right?

Reply:
That for loop is needed for both MOS/FS and FS processing, but the check for primary slit was only needed to make sure that the source position could be assigned for pure FS processing.  Since non-primary slits are never marked point sources by time it gets to the wavecorr step, you can just check for a point source.  MOS/FS point sources already have positions assigned, via the MSA file in assign_wcs.

Reply:
Why are non-primary slits never marked point sources?  I assume this is only true for the FS mode?

Reply:
Yes, it only matters for FS because we have no way of knowing where the source fell for the non-primary slit.  In the srctype step for NIRSpec FS, non-primary sources are always marked EXTENDED because all of the point source handling depends on knowing where the source is.